perf_guide.md

```diff
@@ -67,17 +67,49 @@ addition is 2.7x slower than the one using vertical vector operations!
 
 ## Target-features
 
+Not all processors of a certain architecture will have SIMD processing units,
+and using a SIMD instruction when it's not supported will trigger a CPU exception.
+
+To allow building safe, portable programs, the Rust compiler will **not**, by default,
+generate any sort of vector instructions, unless it can statically determine
+they are supported (for example, on AMD64, SSE2 support is architecturally guaranteed).
+
 ### RUSTFLAGS
 
-TODO
+One of the easiest ways to benefit from SIMD is to allow the compiler
+to generate code using certain vector instruction extensions.
+
+The environment variable `RUSTFLAGS` can be used to pass options for code
+generation to the Rust compiler. These flags will affect **all** compiled crates.
```

Comment:
I would put this last part upfront, since it is probably the key in using Rustflags. 

Maybe something like "`RUSTFLAGS` can be used to enable target features for the whole binary, that is, all crates in the dependency graph will be compiled with the assumption that all CPUs where the binary will run on have these features available". I would also remark that running a binary in a CPU that doesn't have one of these features is undefined behavior. 

And then go to the different ways of enabling features with rustflags. 

There is also a way of specifying the rustflags in the .cargo/config, maybe we should add a TODO at the end of this RUSTFLAGS section to discuss that.